lib/core/messageReceiver.ts

```diff
@@ -476,7 +476,9 @@ export class MessageReceiver extends LinkEntity {
             bMessage.messageId,
             this.name
           );
-          await bMessage.complete();
+          if (!bMessage.delivery.remote_settled) {
```

Comment:
Shouldnt this check move to line 471? Otherwise, we will be logging that we are auto-completing when we dont want to auto-complete an already settled message

Reply:
This branch was out of date from master, I have updated it. So pull before making your changes

Reply:
This way ?
```typescript
if (this.autoComplete && this.receiveMode === ReceiveMode.peekLock && !bMessage.delivery.remote_settled)
```

Yeah, it makes sense. 
I'll just check to make sure I don't face any issues with that.

Reply:
Yes